realm/realm-library/src/main/java/io/realm/RealmAny.java

```diff
@@ -0,0 +1,493 @@
+ * type. Calling a getter method for any particular type, that is not
+ * the same type as the stored value, would raise an exception.
+ * <p>
+ * Our recommendation to handle the RealmAny polymorphism is to write a
+ * switch case around the RealmAny type and its inner value class.
+ * <pre>
+ * <code>
+ *      RealmAny realmAny = aRealmObject.realmAnyAttribute;
+ *
+ *      switch (realmAny.getType()) {
+ *          case OBJECT:
+ *              if (realmAny.getValueClass().equals(DogRealmModel.class)) {
+ *                  DogRealmModel value = realmAny.asRealmModel(DogRealmModel.class);
+ *              }
+ *          case INTEGER:
+ *              performAction(realmAny.asInteger());
+ *              break;
+ *          case BOOLEAN:
+ *              performAction(realmAny.asBoolean());
+ *              break;
+ *          case STRING:
+ *              performAction(realmAny.asString());
+ *              break;
+ *          case BINARY:
+ *              performAction(realmAny.asBinary());
+ *              break;
+ *          case DATE:
+ *              performAction(realmAny.asDate());
+ *              break;
+ *          case FLOAT:
+ *              performAction(realmAny.asFloat());
+ *              break;
+ *          case DOUBLE:
+ *              performAction(realmAny.asDouble());
+ *              break;
+ *          case DECIMAL128:
+ *              performAction(realmAny.asDecimal128());
+ *              break;
+ *          case OBJECT_ID:
+ *              performAction(realmAny.asObjectId());
+ *              break;
+ *          case UUID:
+ *              performAction(realmAny.asUUID());
+ *              break;
+ *          case NULL:
+ *              performNullAction();
+ *              break;
+ *      }
+ * </code>
+ * </pre>
+ * <p>
+ * getValueClass() returns the Java class that represents the inner
+ * value wrapped by the RealmAny instance. If the resulting class is
+ * a realization of {@link io.realm.RealmModel} asRealmModel() can be
+ * called to cast the RealmAny value to a Realm object reference.
+ */
+
+public class RealmAny {
+    @Nonnull
+    private final RealmAnyOperator operator;
+
+    RealmAny(@Nonnull RealmAnyOperator operator) {
+        this.operator = operator;
+    }
+
+    final long getNativePtr() {
+        return this.operator.getNativePtr();
+    }
+
+    /**
+     * Gets the inner type of this RealmAny object.
+     *
+     * @return the inner RealmAnyType
+     */
+    public RealmAnyType getType() {
+        return this.operator.getType();
+    }
+
+    /**
+     * Returns the Java class that represents the inner value wrapped by this RealmAny value.
+     *
+     * @return the class that represents the inner value wrapped by this RealmAny value.
+     */
+    @Nullable
+    public Class<?> getValueClass() {
+        return this.operator.getTypedClass();
+    }
+
+    /**
+     * Creates a new RealmAny with the specified initial value.
```

Comment:
Should we remove the word `initial` here and elsewhere. It implies you can update it, which you cannot?